test/unit/local/steps/dispatch.js

```diff
@@ -0,0 +1,389 @@
+    buffer = new Buffer()
+
+    events = sinon.createStubInstance(SyncState)
+    prep = sinon.createStubInstance(Prep)
+  })
+  afterEach('clean pouch', pouchHelpers.cleanDatabase)
+  after('clean config directory', configHelpers.cleanConfig)
+
+  context('when buffer contains an initial-scan-done event', () => {
+    beforeEach(() => {
+      buffer.push([
+        builders.event().action('initial-scan-done').build()
+      ])
+    })
+
+    it('emits an initial-scan-done event via the emitter', async function () {
+      const dispatchedBuffer = dispatch(buffer, { events, prep, pouch: this.pouch })
+      await dispatchedBuffer.pop()
+
+      should(events.emit).have.been.calledOnce()
+      should(events.emit).have.been.calledWithExactly('initial-scan-done')
+    })
+
+    it('does not call any Prep method', async function () {
+      const dispatchedBuffer = dispatch(buffer, { events, prep, pouch: this.pouch })
+      await dispatchedBuffer.pop()
+
+      shouldHaveCalledOn(prep, [])
+    })
+  })
+
+  context('when buffer contains a scan file event', () => {
+    const filePath = 'foo'
+    const stats = { ino: 1, mtime: new Date(), ctime: new Date() }
+
+    beforeEach(() => {
+      buffer.push([
+        builders.event().action('scan').kind('file').path(filePath).stats(stats).build()
+      ])
+    })
+
+    it('triggers a call to addFileAsync with a file Metadata object', async function () {
+      const metafile = { path: filePath, ino: stats.ino, docType: 'file' }
+
+      const dispatchedBuffer = dispatch(buffer, { events, prep, pouch: this.pouch })
+      await dispatchedBuffer.pop()
+
+      shouldHaveCalledOn(prep, ['addFileAsync'])
+      should(prep.addFileAsync).have.been.calledOnce()
+      should(prep.addFileAsync).have.been.calledWithMatch('local', metafile)
+    })
+  })
+
+  context('when buffer contains a scan directory event', () => {
+    const directoryPath = 'foo'
+    const stats = { ino: 1, mtime: new Date(), ctime: new Date() }
+
+    beforeEach(() => {
+      buffer.push([
+        builders.event().action('scan').kind('directory').path(directoryPath).stats(stats).build()
+      ])
+    })
+
+    it('triggers a call to putFolderAsync with a directory Metadata object', async function () {
+      const metadir = { path: directoryPath, ino: stats.ino, docType: 'folder' }
+
+      const dispatchedBuffer = dispatch(buffer, { events, prep, pouch: this.pouch })
+      await dispatchedBuffer.pop()
+
+      shouldHaveCalledOn(prep, ['putFolderAsync'])
+      should(prep.putFolderAsync).have.been.calledOnce()
+      should(prep.putFolderAsync).have.been.calledWithMatch('local', metadir)
+    })
+  })
+
+  context('when buffer contains a created file event', () => {
+    const filePath = 'foo'
+    const stats = { ino: 1, mtime: new Date(), ctime: new Date() }
+
+    beforeEach(() => {
+      buffer.push([
+        builders.event().action('created').kind('file').path(filePath).stats(stats).build()
```

Comment:
Maybe just `.ino(1)` in case we don't care about the dates?

Reply:
We kinda do care about the dates unfortunately. If they're not present, `metadata.buildFile` will throw an error.

Reply:
Sure, but isn't it the builder's job to set them?

Reply:
Maybe this should be part of the constructor indeed.